### DISCUSSION TITLE
Can I create and write PMTiles file directly in C++?

### DISCUSSION THREAD

**Discussion by qianxiaoming** [protomaps/PMTiles#391] [General]
Can I directly create pmtiles files? I do not want to create mbtiles file first. Is there any API for this? thanks.

**Comment by bdon** [comment]
Yes use http://github.com/felt/tippecanoe to create pmtiles directly

**Reply by qianxiaoming** [reply]
Thank you.  But I have write a library to make tiles from many images at once in C++ and save them to various formats. I want to save tiles in PMTiles format directly.

**Reply by bdon** [reply]
Then you can use the header file https://github.com/protomaps/PMTiles/blob/main/cpp/pmtiles.hpp in your program, that's what tippecanoe uses. though you need to bring your own I/O and compression

**Reply by qianxiaoming** [reply]
Thank you. I will check this header file.
